src/main/java/com/google/devtools/build/lib/analysis/platform/PlatformUtils.java

```diff
@@ -96,25 +96,34 @@ public static Platform getPlatformProto(
       } else {
         properties = spawn.getCombinedExecProperties();
       }
-    } else if (spawn.getExecutionPlatform() != null
-        && !Strings.isNullOrEmpty(spawn.getExecutionPlatform().remoteExecutionProperties())) {
+    } else if (spawn.getExecutionPlatform() != null) {
       properties = new HashMap<>();
-      // Try and get the platform info from the execution properties.
-      try {
-        Platform.Builder platformBuilder = Platform.newBuilder();
+      Platform.Builder platformBuilder = Platform.newBuilder();
+
+      String remoteExecutionProperties = spawn.getExecutionPlatform().remoteExecutionProperties();
+      if (remoteExecutionProperties.isEmpty()) {
```

Comment:
Can you reverse this? `if (!remoteExecutionProperties.isEmpty()) { handle remote } else { try exec properties }`

Reply:
Reversed in https://github.com/bazelbuild/bazel/pull/19792/commits/0712177b0335b3591d146a1a70062eec56f2366e